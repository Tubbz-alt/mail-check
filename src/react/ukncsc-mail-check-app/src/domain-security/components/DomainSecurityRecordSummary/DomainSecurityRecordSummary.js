import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import map from 'lodash/map';
import toLower from 'lodash/toLower';
import { Message } from 'semantic-ui-react';
import {
  DomainSecurityRecord,
  DomainSecuritySummaryMessages,
} from 'domain-security/components';
import { DomainSecurityContext } from 'domain-security/context';

const DomainSecurityRecordSummary = ({
  type,
  description,
  id,
  loading,
  error,
  records,
  inheritedFrom,
  failures,
  warnings,
  inconclusives,
}) => (
  <React.Fragment>
    <p>{description}</p>
    {map(records, ({ record }, i) => (
      <DomainSecurityRecord key={i} inheritedFrom={inheritedFrom}>
        {record}
      </DomainSecurityRecord>
    ))}
    {!loading &&
      !error &&
      (records ? (
        <React.Fragment>
          <p>
            <DomainSecurityContext.Consumer>
              {value => (
                <Link to={`/${value}/${id}/${toLower(type)}`}>
                  View more information
                </Link>
              )}
            </DomainSecurityContext.Consumer>
          </p>
          <DomainSecuritySummaryMessages
            id={id}
            type={type}
            failures={failures.length}
            warnings={warnings.length}
            inconclusives={inconclusives.length}
          />
        </React.Fragment>
      ) : (
        <Message info>
          {`We don't have any information about the ${type} record of this domain
          yet. Please check back later or get in touch if you think there's a
          problem.`}
        </Message>
      ))}
    {error && (
      <Message error>
        There was a problem retrieving the {type} summary for this domain:{' '}
        {error.message}
      </Message>
    )}
  </React.Fragment>
);

DomainSecurityRecordSummary.defaultProps = {
  id: null,
  loading: false,
  error: null,
  records: null,
  inheritedFrom: null,
  failures: [],
  warnings: [],
  inconclusives: [],
};

DomainSecurityRecordSummary.propTypes = {
  type: PropTypes.oneOf(['DMARC', 'SPF', 'TLS']).isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.string,
  loading: PropTypes.bool,
  error: PropTypes.shape({ message: PropTypes.string }),
  records: PropTypes.arrayOf(PropTypes.shape({ record: PropTypes.string })),
  inheritedFrom: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
  }),
  failures: PropTypes.arrayOf(PropTypes.string),
  warnings: PropTypes.arrayOf(PropTypes.string),
  inconclusives: PropTypes.arrayOf(PropTypes.string),
};

export default DomainSecurityRecordSummary;
