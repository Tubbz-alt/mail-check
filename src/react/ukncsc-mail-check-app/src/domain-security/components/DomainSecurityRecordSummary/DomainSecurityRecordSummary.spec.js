import React from 'react';
import { render } from 'react-testing-library';
import DomainSecurityRecordSummary from './DomainSecurityRecordSummary';

describe('DomainSecurityRecordSummary', () => {
  let container;

  describe('when no records are provided', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityRecordSummary type="SPF" description="bleh" />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should not display a record', () =>
      expect(
        container.getElementsByClassName('DomainSecurityRecord')
      ).toHaveLength(0));
  });
});
