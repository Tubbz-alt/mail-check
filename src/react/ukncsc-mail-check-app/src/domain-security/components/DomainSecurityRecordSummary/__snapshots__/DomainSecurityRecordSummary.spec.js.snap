// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityRecordSummary when no records are provided it should match the snapshot 1`] = `
<div>
  <p>
    bleh
  </p>
  <div
    class="ui info message"
  >
    We don't have any information about the SPF record of this domain yet. Please check back later or get in touch if you think there's a problem.
  </div>
</div>
`;
